compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -5078,3 +5079,76 @@ declare_lint! {
     };
     crate_level_only
 }
+
+declare_lint! {
+    /// The `abi_unsupported_vector_types` lint detects function definitions and calls
+    /// whose ABI depends on enabling certain target features that do not enable those features.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,ignore (fails on non-x86_64)
+    /// #![allow(improper_ctypes_definitions)] // false positive
+    ///
+    /// pub extern "C" fn foo(_: std::arch::x86_64::__m256) {
+    ///   todo!()
+    /// }
+    ///
+    /// #[target_feature(enable = "avx")]
+    /// pub unsafe extern "C" fn favx(_: std::arch::x86_64::__m256) {
+    ///   todo!()
+    /// }
+    ///
+    /// # #[target_feature(enable = "avx")]
+    /// # unsafe fn helper() {
+    /// #   foo(unsafe { std::mem::zeroed() });
+    /// # }
+    /// # fn main() {
+    /// #   let v = unsafe { std::mem::zeroed() };
+    /// #   unsafe { favx(v); }
```

Comment:
This line is hidden from the docs and yet it is a line that triggers the lint -- that doesn't make sense to me?

Reply:
Not sure what I was going for here :-) should have fixed it now